From cced21a7c4744e0786e11c9fd2d6c2b21b6f89f2 Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Sun, 3 Jul 2016 20:04:55 +0200
Subject: [PATCH] Do not adjust display mode to fps in set_vout_mode()

Adjusting display mode to fps doesn't work well if we want to switch display mode during
a video playback. For example, if we'll try to switch to 1080p24hz mode from 1080p60hz mode
during a playback of a 23.976 FPS video, we'll get no signal on the display.

Original patch by @codesnake adjusted for 3.14 kernel: 5a582da01756f185d49d5943cf6690fbde5be20f
---
 drivers/amlogic/display/vout/tv_vout.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/display/vout/tv_vout.c b/drivers/amlogic/display/vout/tv_vout.c
index 57ca080..9148420 100644
--- a/drivers/amlogic/display/vout/tv_vout.c
+++ b/drivers/amlogic/display/vout/tv_vout.c
@@ -753,8 +753,8 @@ static int tv_set_current_vmode(enum vmode_e mode)
 		if (DOWN_HPLL == fine_tune_mode)
 			update_tv_info_duration(fps_target_mode, UP_HPLL);
 	}
-	fps_auto_adjust_mode(&tvmode);
-	update_vmode_status(get_name_from_vmode(tvmode));
+	// fps_auto_adjust_mode(&tvmode);
+	// update_vmode_status(get_name_from_vmode(tvmode));
 	vout_log_info("%s[%d]fps_target_mode=%d\n",
 		      __func__, __LINE__, tvmode);
 
-- 
1.8.3.1

